### PR TITLE
[BEAM-115] Port fn_api_runner to be able to use runner protos.

### PR DESCRIPTION
**Pull request by robertwb**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`.
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by robertwb** [comment]
R: @vikkyrk

File: sdks/python/apache_beam/runners/worker/sdk_worker.py [hunk 1]
```diff
@@ -296,6 +319,51 @@ def initial_source_split(self, request, unused_instruction_id=None):
     return response
 
   def create_execution_tree(self, descriptor):
+    if descriptor.primitive_transform:
```
Comment (vikkyrk) [inline sdks/python/apache_beam/runners/worker/sdk_worker.py]: I forget, why can't we memoize here based on `ProcessBundleDescriptor.id`?

File: sdks/python/apache_beam/runners/worker/sdk_worker.py [hunk 2]
```diff
@@ -296,6 +319,51 @@ def initial_source_split(self, request, unused_instruction_id=None):
...
+        descriptor, self.data_channel_factory, counter_factory, state_sampler,
+        self.state_handler)
+
+    pcoll_consumers = collections.defaultdict(list)
+    for transform_id, transform_proto in descriptor.transforms.items():
+      for pcoll_id in transform_proto.inputs.values():
+        pcoll_consumers[pcoll_id].append(transform_id)
+
+    @memoize
+    def get_operation(transform_id):
```
Comment (vikkyrk) [inline sdks/python/apache_beam/runners/worker/sdk_worker.py]: With memoization of operations, is the execution tree thread-safe across parallel `process_bundle` requests? I also see `DataOutputOperation` being mutated to set the `instruction_id`, so wondering if it is safe.

Comment (robertwb) [inline sdks/python/apache_beam/runners/worker/sdk_worker.py]: The memoization is local to the execution of this bundle. Eventually we'll want to cache and re-used these (though not concurrently).

Comment (vikkyrk) [inline sdks/python/apache_beam/runners/worker/sdk_worker.py]: Ah right.

**Comment by vikkyrk** [comment]
LGTM, after branch conflict fixes.

**Pull request closed without merging**